src/main/java/clueless/Game.java

```diff
@@ -213,12 +204,26 @@ public boolean hasPlayer() {
 	 */
 	public Player whoHasClue(List<Card> suggestion) {
 		Player pl = null;
```

Comment:
its probably best to remove this and just return (Player) this.characterMap.get(CHARACTER_TURN_ORDER[i]) directly

we could get caught creating a player object with every whoHasClue method call which can lead to a memory leak. This might not happen if we're 100% confident that you setting p1 to the player really does act as a pointer/link and not as a copied object. But I am not really willing to make that call and would just prefer to return the Player directly from the map to avoid the risk altogether.